Makefile

```diff
@@ -28,19 +34,48 @@ ${PROJECT_NAME}.json: ${PROJECT_NAME}.v
 	yosys -p "synth_xilinx -flatten -abc9 -nobram -arch xc7 -top ${PROJECT_NAME}; write_json ${PROJECT_NAME}.json" $<
 
 ${PROJECT_NAME}.fasm: ${PROJECT_NAME}.json
-	nextpnr-xilinx --chipdb ${CHIPDB_DIR}/${PART}.bin --xdc ${PROJECT_NAME}-${BOARD}.xdc --json $< --write ${PROJECT_NAME}_routed.json --fasm $@ --verbose --debug
+	${NEXTPNR_DIR}/bin/nextpnr-xilinx --chipdb ${CHIPDB_DIR}/${PART}.bin --xdc ${PROJECT_NAME}-${BOARD}.xdc --json $< --write ${PROJECT_NAME}_routed.json --fasm $@ --verbose --debug
 
 ${PROJECT_NAME}.frames: ${PROJECT_NAME}.fasm
-	@. "${XRAY_DIR}/utils/environment.sh"
-	fasm2frames --part ${PART} --db-root ${DB_DIR}/kintex7 $< > $@
+	prjxray/env/bin/fasm2frames --part ${PART} --db-root ${DB_DIR}/kintex7 $< > $@
```

Comment:
Change to `${XRAY_DIR}/env/bin/fasm2frames --part ${PART} --db-root ${DB_DIR}/kintex7 $< > $@` ?

Reply:
env/bin does not exist in the installed XRAY_DIR location (~/opt/prjxray) - it only exists in the local build location (./prjxray), so such a change would break the build.
Is there a way to make fasm2frames get installed into the installed XRAY_DIR location (~/opt/prjxray) ?

Reply:
I will add XRAY_UTILS_DIR and XRAY_TOOLS_DIR variables. The former points to the local build area, the latter points to the installed location.

Reply:
I think we've had different scenarios in mind.. I had been assuming that an independent clone of prjxray be available in `XRAY_DIR=~/opt/prjxray`, i.e. cloning prjxray would not be controlled through this Makefile. This way, we would have both the `XRAY_UTILS_DIR` and `XRAY_TOOLS_DIR` in the same location. Either way, I think we don't need `make install` for our use of prjxray.

I don't know if fasm2frames can get installed to `XRAY_DIR`. In fact, make install seems to only be concerned with
- $INSTALL_DIR/usr/local/bin/xc7frames2bit
- $INSTALL_DIR/usr/local/bin/xc7patch
- $INSTALL_DIR/usr/local/bin/gen_part_base_yaml
- $INSTALL_DIR/usr/local/bin/frame_address_decoder
- $INSTALL_DIR/usr/local/bin/bittool
- $INSTALL_DIR/usr/local/bin/segmatch
- $INSTALL_DIR/usr/local/bin/bitread

Reply:
Yes, I wonder why one is installed and one is left in the build area.  If we need to use fasm2frames for this project, then one would expect others to also need to use it.  Or is there some other way that prjxray is intended to be used which does not require fasm2frames to be called independently?

Reply:
There is a related discussion from 2020 at https://github.com/f4pga/prjxray/issues/1384 that resulted in an independent repo for `fasm2frames`: https://github.com/SymbiFlow/f4pga-xc-fasm I have not tested the code therein.

I opened an issue for further clarification at https://github.com/SymbiFlow/f4pga-xc-fasm/issues/19

Reply:
The location of fasm2frames seems to be the only (and seemingly unresolved) open question at this time. Why don't you send the pull request to update the make process at kintex-chatter, and we file an issue against https://github.com/rwhitby/xc7k325t-blinky-nextpnr/pull/1/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R45 to track and resolve the need for prjxray sources in the future?

Reply:
Opened https://github.com/kintex-chatter/xc7k325t-blinky-nextpnr/pull/25 with these notes included.